drv/i2c-devices/src/max5970.rs

```diff
@@ -240,6 +290,78 @@ impl Max5970 {
     pub fn i2c_device(&self) -> &I2cDevice {
         &self.device
     }
+
+    fn convert_volts(&self, mon_range: MonRange, msb: u8, lsb: u8) -> Volts {
+        //
+        // The 10-bit value from the ADC is a fraction of the full-scale
+        // voltage setting.
+        //
+        let divisor = 1024.0 / mon_range.full_scale_voltage(self.rail) as f32;
+
+        Volts(((((msb as u16) << 2) | (lsb as u16)) as f32) / divisor)
+    }
+
+    fn convert_current(
+        &self,
+        status2: Status2,
+        msb: u8,
+        lsb: u8,
+    ) -> Result<Amperes, ResponseCode> {
+        let millivolts = status2
+            .max_current_sense_range(self.rail)
+            .ok_or(ResponseCode::BadDeviceState)?;
+
+        //
+        // The 10-bit value from the ADC is a fraction of the maximum
+        // current-sense range.
+        //
+        let divisor = 1024.0 / millivolts as f32;
+        let delta = ((((msb as u16) << 2) | (lsb as u16)) as f32) / divisor;
+
+        //
+        // We have the voltage drop across the current sense resistor; to
+        // determine current, we divide voltage by resistance (I = V / R).
+        //
+        Ok(Amperes(delta / self.rsense as f32))
+    }
+
+    pub fn max_vout(&self) -> Result<Volts, ResponseCode> {
+        let (msb, lsb) = if self.rail == 0 {
+            (
+                self.read_reg(Register::max_chx_mon_msb_ch1)?,
+                self.read_reg(Register::max_chx_mon_lsb_ch1)?,
+            )
+        } else {
+            (
+                self.read_reg(Register::max_chx_mon_msb_ch2)?,
+                self.read_reg(Register::max_chx_mon_lsb_ch2)?,
+            )
+        };
+
+        let mon_range = MonRange(self.read_reg(Register::mon_range)?);
+        Ok(self.convert_volts(mon_range, msb, lsb))
+    }
+
+    pub fn max_iout(&self) -> Result<Amperes, ResponseCode> {
+        let (msb, lsb) = if self.rail == 0 {
+            (
+                self.read_reg(Register::max_chx_cs_msb_ch1)?,
+                self.read_reg(Register::max_chx_cs_lsb_ch1)?,
+            )
+        } else {
+            (
+                self.read_reg(Register::max_chx_cs_msb_ch2)?,
+                self.read_reg(Register::max_chx_cs_lsb_ch2)?,
+            )
+        };
```

Comment:
```suggestion
        let (msb_reg, lsb_reg) = if self.rail == 0 {
            (Register::max_chx_cs_msb_ch1, Register::max_chx_cs_lsb_ch1)
        } else {
            (Register::max_chx_cs_msb_ch2, Register::max_chx_cs_lsb_ch2)
        };
        let msb = self.read_reg(msb_reg)?;
        let lsb = self.read_reg(lsb_reg)?;
```

Reply:
(You could also change the existing `read_vout` and `read_iout` implementations to match, even though they aren't touched in this PR)